source/yats/include/yats/task_configurator.h

```diff
@@ -59,13 +59,16 @@ class abstract_task_configurator
     }
 };
 
-template <typename Task>
+template <typename Task, typename... Parameters>
 class task_configurator : public abstract_task_configurator
 {
 public:
     using helper = decltype(make_helper(&Task::run));
 
-    task_configurator() = default;
+    task_configurator(Parameters&&... parameters)
+        : m_construction_parameters(std::forward_as_tuple(parameters...))
```

Comment:
Forwarding rvalue references makes it impossible to use a pipeline twice for the construction of a scheduler because the first construction will consume the variable.

Reply:
It is possible to to keep this behavior but it requieres to change the constructor of the scheduler to take the pipeline by value so the user needs to move it into the Scheduler.